src/services/review/ReviewRequestService.ts

```diff
@@ -0,0 +1,365 @@
+  ): Promise<ReviewRequestDto | RequestNotFoundError> => {
+    const { siteName } = userWithSiteSessionData
+    const review = await this.repository.findOne({
+      where: {
+        siteId: site.id,
+      },
+      include: [
+        {
+          model: ReviewMeta,
+          as: "reviewMeta",
+          where: {
+            pullRequestNumber,
+          },
+        },
+        {
+          model: User,
+          as: "requestor",
+        },
+        {
+          model: User,
+          as: "reviewers",
+        },
+        {
+          model: Site,
+        },
+      ],
+    })
+
+    // As the db stores github's PR # and (siteName, prNumber)
+    // should be a unique identifier for a specific review request,
+    // unable to find a RR with the tuple implies that said RR does not exist.
+    // This could happen when the user queries for an existing PR that is on github,
+    // but created prior to this feature rolling out.
+    if (!review) {
+      return new RequestNotFoundError()
+    }
+
+    // NOTE: We explicitly destructure as the raw data
+    // contains ALOT more than these fields, which we want to
+    // discard to lower retrieval times for FE
+    const { title, created_at } = await this.apiService.getPullRequest(
+      siteName,
+      pullRequestNumber
+    )
+
+    const changedItems = await this.compareDiff(userWithSiteSessionData)
+
+    return {
+      reviewUrl: review.reviewMeta.reviewLink,
+      title,
+      status: review.reviewStatus,
+      requestor: review.requestor.email || "",
+      reviewers: review.reviewers.map(({ email }) => email || ""),
+      reviewRequestedTime: new Date(created_at).getTime(),
+      changedItems,
+    }
+  }
+
+  /**
+   * Updates the review request with provided details.
+   * Note that the semantics for updating the review request description
+   * is as follows:
+   *
+   * 1. If the body is `undefined`, we **do not** update it.
+   * 2. If the body is `""` (an empty string), we update the\
+   * github pull request description to be empty too.
+   * 3. Otherwise, we just write through to github.
+   */
+  updateReviewRequest = async (
+    reviewRequest: ReviewRequest,
+    { title, description, reviewers }: RequestChangeInfo
+  ) => {
+    // Update db state with new reviewers
+    reviewRequest.reviewers = reviewers
+    const savePromise = reviewRequest.save()
+
+    const siteName = reviewRequest.site.name
+    const { pullRequestNumber } = reviewRequest.reviewMeta
+
+    // Update github state
+    const githubUpdatePromise = this.apiService.updatePullRequest(
+      siteName,
+      pullRequestNumber,
+      title,
+      description
+    )
+
+    await Promise.all([savePromise, githubUpdatePromise])
+  }
+
+  // NOTE: The semantics of our reviewing system is slightly different from github.
+  // The approval is tied to the request, rather than the user.
+  approveReviewRequest = async (reviewRequest: ReviewRequest) => {
+    reviewRequest.reviewStatus = "APPROVED"
```

Comment:
Should we use an enum instead? Same for the other `reviewStatus` below.

Reply:
No strong opinion here, as functionality is the same. I can change it to enum!

Reply:
Yup functionally is the same, but I think having an enum helps to enforce correctness a little bit better.